lambdas/importer.ts

```diff
@@ -236,33 +235,43 @@ async function importEventsToDynamoDb(
 			},
 			body: requestBody,
 		};
-
 		try {
 			const response = await fetch(meetupGraphQlEndpoint, requestOptions);
 			const res = (await response.json()) as MeetupFutureEventsPayload;
 			const unifiedEvents = res.data.events?.unifiedEvents;
 			const events =
 				unifiedEvents?.edges.map((edge) => {
-					edge.node.dateTime = new Date(edge.node.dateTime); // Rewrite string timestamp to Date object
+					edge.node.dateTime = new Date(edge.node.dateTime);
+					const currentDate = new Date(); // Rewrite string timestamp to Date object
+					currentDate.setMonth(currentDate.getMonth() + 6);
+					//console.log(events);
 					return edge.node;
 				}) ?? [];
-
-			if (unifiedEvents?.pageInfo.hasNextPage) {
-				const nextCursor = unifiedEvents.pageInfo.endCursor;
-				const nextEvents = await fetchAllFutureEvents(
-					urlname,
-					nextCursor,
-				);
-				events.push(...nextEvents);
+			events.forEach((events) => {
+				//check if event is within 6 months
+				const currentDate = new Date(); // Rewrite string timestamp to Date object
+				currentDate.setMonth(currentDate.getMonth() + 6); //set date to 6 months from now
+				if (events.dateTime >= currentDate) {
+					done = true;
+				}
+			});
+			if (!done) {
+				if (unifiedEvents?.pageInfo.hasNextPage) {
+					const nextCursor = unifiedEvents.pageInfo.endCursor;
+					const nextEvents = await fetchAllFutureEvents(
+						urlname,
+						nextCursor,
+					);
+					events.push(...nextEvents);
+				}
 			}
```

Comment:
I think something like this would work and is a lot simpler
```suggestion
			const currentDate = new Date();
			currentDate.setMonth(currentDate.getMonth() + 6);
			const hasEventsPast6Months = events.some(
				(event) => event.dateTime >= currentDate,
			);

			if (!hasEventsPast6Months && unifiedEvents?.pageInfo.hasNextPage) {
				const nextCursor = unifiedEvents.pageInfo.endCursor;
				const nextEvents = await fetchAllFutureEvents(
					urlname,
					nextCursor,
				);
				events.push(...nextEvents);
			}
```

Reply:
Note: I did not test this lol. Could you double check that it actually worked